File: C_Programming/Study/Q4_Assignment11.c
#include <stdio.h>

///////////////////////////////////////////////////////////////////////////////////////////////////////////
//
// Function Name :  RangeSumEven
// Output:          
// Description:     Assignment_11: Q.4  : Write a program which accept Range from user & return addition of all Even numbers  
//                                        in between that range. (Range should contains positive numbers only)    
// Date:            30/05/2026
// Author:          Shital Ajit Nikam   
//
///////////////////////////////////////////////////////////////////////////////////////////////////////////


int RangeSumEven(int iStart,int iEnd)
{
    int iCnt = 0;
    int iSum = 0;
       
    //Updater Used
    if(iStart < 0 || iEnd < 0 || iEnd < iStart)
    {
        return -1;
    }
    
    for(iCnt = iStart; iCnt <= iEnd; iCnt++)
       {
            if(iCnt % 2 == 0)
            {
                iSum = iSum + iCnt; 
            }
                            
       }
    return iSum;
}

int main()
{
    int iValue1 = 0, iValue2 = 0;
    int iRet = 0;
        
    printf("Enter Starting point: \n");
    scanf("%d", &iValue1);

    printf("Enter ending point: \n");
    scanf("%d", &iValue2);

    iRet = RangeSumEven(iValue1,iValue2);
    if(iRet == -1)
    {
        printf("Invalid Range \n");
    }
    else
    {
        printf("Addition of Even number is: %d", iRet);
    }
    

    return 0;
}

/////////////////////////////////////////////////////////////////////////////////////
//Time Complexity : O(N)  where N = Range of that number
/////////////////////////////////////////////////////////////////////////////////////
// Input : 23 30
// Output : 108 
// Input : 10 18
// Output : 70
// Input : -10 2
// Output : Invalid Range
// Input : 90 18
// Output : Invalid Range
// 
/////////////////////////////////////////////////////////////////////////////////////

File: Python_Programming/Competative/Q4_Assignment11.py
#///////////////////////////////////////////////////////////////////////////////////////////////////////////////////
# Function Name :  RevNumber
#        
# Description:     Assignment_11: Q.4: Write a program which accepts one number & print reverse of that number. 
#                                      
# Input  :         123
# Output :         321
#                                     
# Date:            24/05/2026
# Author:          Shital Ajit Nikam   
#
#///////////////////////////////////////////////////////////////////////////////////////////////////////////////////

def RevNumber(No):
    i = 0
    Rev = 0
    Digit = 0 
    if(No < 0):
        No = -No
        
    if(No == 0):
        return 0  
         
    while(No != 0):   
        Digit = No % 10                         # Seperate the Digit
        Rev = (Rev * 10) + Digit                # Reverse the digits             
        No = No // 10                   
                    
    return Rev;              
    
def main():
    value = int(input("Enter the number: "))
        
    Ret = RevNumber(value) 
    print("Reverse number is : ",Ret)
               
    
if __name__ == "__main__":
    main()